src/Server.zig

```diff
@@ -469,6 +470,101 @@ fn getAstCheckDiagnostics(
     }
 }
 
+/// TODO: Memory
+pub fn getFullCompilationDiagnostics(
+    server: *Server,
+    handle: DocumentStore.Handle,
+    diagnostics: *std.ArrayListUnmanaged(types.Diagnostic),
+) !void {
+    var allocator = server.arena.allocator();
+    const config = server.config;
+
+    const build_file_uri = handle.associated_build_file orelse return;
+    const build_file = server.document_store.build_files.get(build_file_uri).?;
+
+    const tracy_zone = tracy.trace(@src());
+    defer tracy_zone.end();
+
+    var arena = std.heap.ArenaAllocator.init(allocator);
+    defer arena.deinit();
+    const arena_allocator = arena.allocator();
+
+    const build_file_path = try uri_utils.parse(arena_allocator, build_file.uri);
+    const directory_path = try std.fs.path.resolve(arena_allocator, &.{ build_file_path, "../" });
+
+    // TODO extract this option from `BuildAssociatedConfig.BuildOption`
+    const zig_cache_root: []const u8 = try std.fs.path.join(arena_allocator, &.{ directory_path, "zig-cache" });
+    // Since we don't compile anything and no packages should put their
+    // files there this path can be ignored
+    const zig_global_cache_root: []const u8 = "ZLS_DONT_CARE";
+
+    const standard_args = [_][]const u8{
+        config.zig_exe_path.?,
+        "run",
+        config.diagnostics_build_runner_path.?,
+        "--cache-dir",
+        config.global_cache_path.?,
+        "--pkg-begin",
+        "@build@",
+        build_file_path,
+        "--pkg-end",
+        "--",
+        config.zig_exe_path.?,
+        directory_path,
+        zig_cache_root,
+        zig_global_cache_root,
+    };
+
+    const arg_length = standard_args.len + if (build_file.build_associated_config) |cfg| if (cfg.build_options) |options| options.len else 0 else 0;
```

Comment:
this will need a way to pass:
- `-fno-emit-bin`
- custom `build.zig` arguments for diagnostics as a config option for zls.json

Reply:
[Already handled](https://github.com/zigtools/zls/pull/871/files#diff-2b8e6e0157f1a82b0d7b30ff44364d62044cb3213f4ad91df0bf5bd72fb04983R237-R244) (diagnostics build runner, line 237-244)